lib/model/EcoscoreData.dart

```diff
@@ -27,4 +27,7 @@ class EcoscoreData extends JsonObject {
 
   @override
   Map<String, dynamic> toJson() => _$EcoscoreDataToJson(this);
+
+  static Map<String, dynamic> toJsonHelper(EcoscoreData d) =>
+      d != null ? d.toJson() : {};
```

Comment:
Given that the `ecoscore_data` field is only translated to JSON when it is different from `null` them "empty map" case should never occur:

```dart
writeNotNull(
      'ecoscore_data', EcoscoreData.toJsonHelper(instance.ecoscoreData));
```
... where `writeNotNull` is defined as follows:
```dart
writeNotNull(
      'ecoscore_data', EcoscoreData.toJsonHelper(instance.ecoscoreData));
```
So even though we change it to return `null` rather than the empty map it would not make a difference.

So I think it is fine to keep it as it is.

Reply:
As a site note, I deleted my previous comment because I misunderstood but you were saying to begin with :) Sorry for the confusion. Anyway, I'll mark this as resolved. Let me know if you disagree.